music/src/music_rnn/model.ts

```diff
@@ -226,7 +244,10 @@ export class MusicRNN<A extends controls.ControlSignalUserArgs> {
       }
 
       if (splitControls) {
-        nextInput = nextInput.concat(splitControls[i + 1], 1);
+        nextInput = tf.concat([splitControls[i + 1], nextInput], 1);
```

Comment:
We should do a single concat for performance reasons.
You should build up the array here and then call concat once.

Reply:
Done.